tools/lsp/preview.rs

```diff
@@ -124,6 +125,37 @@ pub fn poll_once<F: std::future::Future>(future: F) -> Option<F::Output> {
     }
 }
 
+fn invalidate_contents(url: &lsp_types::Url) {
+    i_slint_core::debug_log!("PREVIEW: Invalidate {url}");
+    let component = {
+        let cache = CONTENT_CACHE.get_or_init(Default::default).lock().unwrap();
+        (cache.dependency.contains(url) && cache.ui_is_visible)
+            .then_some(())
+            .and_then(|_| cache.current_component())
+    };
+
+    if let Some(document_cache) = document_cache() {
+        // Do not add random things to the docuemnt cache!
+        if document_cache.get_document(url).is_some() {
+            let dc = document_cache.snapshot();
```

Comment:
snapshot is very expensive, why do we do it there?

Reply:
Because the snapshot gives me a mutable document_cache. The "normal" one is not.

Reply:
I removed the snapshot now as there is no real need to update the DocumentCache at all.